proposals/repo-to-tar/README.md

```diff
@@ -0,0 +1,102 @@
+
+```md
+├── blobs
+│   └── sha256
+│       ├── 45e16698e6e721de091b5ecbb811eae1243fb8f2ed5dbd04a9f40cd199c355f9
+│       ├── 4da097607a3cb58407cd8c157271bb5f7462a5872d516a5bc979ec678ac30f7d
+│       └── 5c2dafe3c70c13990190d643c91e9f67b8129b179257674888178868474f6511
+├── index.json
+└── oci-layout
+```
+where one of the files in sha256 is the actual file tar and others 2 are configs and manifests.<br>
+
+2. Add a flag `--to-oci-tar` to `imgpkg push`  to be able to creata a tar while pushing a tar with imgpkg push command.
+ 
+3. Add a flag `--oci-tar` to `imgpkg copy` to be able to copy this oci kind of tar to a registry and differentiate with the `--tar` flag.
+
+#### Future Goals
+
+1. An improvement would be to have only one flag `--tar` for `imgpkg copy` making no changes to flags but making it compatible and being smart enough to understand and differentiate between both kind of tars without any need for specification. <br>
+
+2. Add an inflate option to also contain all refs images in the oci-tar as discussed [here](https://github.com/carvel-dev/imgpkg/issues/55#issuecomment-962209740). This will be useful and true to the name of the command `imgpkg copy` as it will copy all the refs images as well and create a self-sufficient oci-tar. <br>
+
+### Specification / Use Cases
+
+Use Cases : 
+- User wants to build a local oci compatible tar that can be pushed at a later point without being dependent on a registry.
+- While using a CI framework which allows users to share files between stages, user wants to pass this output so that it can be pushed/copied in the next stage rather than having a shared registry.
+- User can do the above with Package and PackageRepository bundles
+
+### Other Approaches Considered
+
+The idea behind the usage of `imgpkg push -b my.registry.io/some-name -f some-folder --to-oci-tar local-oci-format.tar` is to concentrate in a single command all the options to create a new image, that being to the registry or directly to disk.
+
+I had considered not do the above but modifying the `imgpkg push command` to not require `-f folder` flag and being compatible with
+`imgpkg push -b my.registry.io/some-name --to-oci-tar local-oci-format.tar` to push the tar to the registry. <br><br>
+
+The reason this was dropped was it is similar to imgpkg copy command used above, although initially I had considered it because the copy command can be misleading when a user just wants to push his oci-tar. <br><br>
+
+
+## Open Questions
+1. Do we need save command to be able to save a tar file from a bundle image or should we just use push command to do the same as it forms the tar ? <br> 
+2. If save can create a `oci-tar` and `copy` can copy/push the tar to the repo, what is the need to make it compatible with `push` command ? <br>
+3. Should we have a single flag `--tar` for `imgpkg copy` to be able to copy both kind of tars or should we have 2 flags `--tar` and `--oci-tar` ? <br>
```

Comment:
In the future we can decide to make the --tar flag smart enough, but does not need to be part of the MVP for this feature

Reply:
HI, specified it in the answers to the open questions. Also, added a line in goals to emfasize that imgpkg push --to-oci-tar will only create in disk the OCI image for the bundle itself and inflate option in future goals.

Reply:
If the above questions are already answered let us remove them from the Open questions section so that we can move the proposal to Review state

Reply:
Shall I remove open as well as answered questions section or maybe we can keep them for future reference since they are answered in the answered section ? What is the correct standard to follow ?

Reply:
we can keep the section empty

Reply:
Made changes